x-pack/plugins/uptime_monitoring/public/components/queries/ping_list/ping_list.tsx

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  EuiBadge,
+  EuiComboBox,
+  EuiComboBoxOptionProps,
+  EuiEmptyPrompt,
+  EuiFieldNumber,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiHealth,
+  // @ts-ignore
+  EuiInMemoryTable,
+  EuiLoadingChart,
+  EuiPanel,
+  EuiTitle,
+  EuiToolTip,
+} from '@elastic/eui';
+import { get } from 'lodash';
+import moment from 'moment';
+import React, { Fragment } from 'react';
+import { Query } from 'react-apollo';
+import { Ping } from 'x-pack/plugins/uptime_monitoring/common/graphql/types';
+import { UMPingSortDirectionArg } from '../../../../common/domain_types';
+import { getPingsQuery } from './get_pings';
+
+interface PingListProps {
+  monitorId?: string;
+  dateRangeStart: number;
+  dateRangeEnd: number;
+  autorefreshInterval: number;
+  autorefreshEnabled: boolean;
+  sort?: UMPingSortDirectionArg;
+  size?: number;
+}
+
+interface PingListState {
+  statusOptions: EuiComboBoxOptionProps[];
+  selectedOption: EuiComboBoxOptionProps;
+  maxSearchSize: number;
+}
+
+export class Pings extends React.Component<PingListProps, PingListState> {
```

Comment:
Just a pattern I have found useful... props and state on any given component can easily just be named ComponentProps and ComponentState vs explicitly naming them. Makes refactors later much easier. Again just a comment not an issue

Reply:
I think a lot of these props will end up getting refactored into a `CommonProps` interface because most of the components used in this plugin so far have the exact same props.

So props will end up being `CommonProps | ComponentProps` if it has specific props it needs outside of those. I'll consider updating that in the future though.